airflow/www/views.py

```diff
@@ -924,6 +926,56 @@ def rendered(self):
             title=title,
         )
 
+    @expose('/rendered-k8s')
+    @auth.has_access(
+        [
+            (permissions.ACTION_CAN_READ, permissions.RESOURCE_DAG),
+            (permissions.ACTION_CAN_READ, permissions.RESOURCE_TASK_INSTANCE),
+        ]
+    )
+    @action_logging
+    def rendered_k8s(self):
+        """Get rendered k8s yaml."""
+        k8s = is_k8s_or_k8scelery_executor()
+        if not k8s:
+            abort(404)
+        dag_id = request.args.get('dag_id')
+        task_id = request.args.get('task_id')
+        execution_date = request.args.get('execution_date')
+        dttm = timezone.parse(execution_date)
+        form = DateTimeForm(data={'execution_date': dttm})
+        root = request.args.get('root', '')
+        logging.info("Retrieving rendered templates.")
+        dag = current_app.dag_bag.get_dag(dag_id)
+        task = copy.copy(dag.get_task(task_id))
+        ti = models.TaskInstance(task=task, execution_date=dttm)
+        try:
+            pod_spec = ti.render_k8s_pod_yaml()
```

Comment:
Shouldn't this be using `RenceredTIFIelds.get_k8s_pod_yaml`?

Or if not, we should delete that function.

Reply:
@ashb how come? It requires metadata in the task instance to generate the pod.

Reply:
Much like Rendered TaskInstance fields - once it's run, the db holds exactly was was used (rather than the current rendering)

But either way if we aren't using that col for display: why are we storing it at all?

Reply:
Can we generate this pod spec just off what is stored in a Serialised DAG/Task?

Reply:
@ashb we store it because the values are different between task runs, so if someone goes to a historical DAG run they should see what it looked like at the time of running.

Reply:
But where do we read it?

Reply:
> if someone goes to a historical DAG run they should see what it looked like at the time of running.

I agree with this

Reply:
But where do we use `RenderedTaskInstanceFields.get_k8s_pod_yaml` method?

Reply:
@ryanahamilton do we need to utilize this somehow?

@ashb we could also just merge this for the beta and figure that out later. It's not a breaking issue and it's a functionality we want.